packages/core/src/elements/FormElement.ts

```diff
@@ -0,0 +1,27 @@
+import type { SourceConfiguration } from '../SourceConfiguration'
+import type { BaseElementEvents } from '../utils/types'
+
+import { SourceElement } from './SourceElement'
+
+export interface FormEvents extends BaseElementEvents {
+  /**
+   * Fired when a form response is submitted
+   */
+  submitted: { id: string }
```

Comment:
I know this pattern was already set in scheduler so I'm fine with keeping this for now - but what are your thoughts around using GraphQL typegen to generate types off of fragments in this codebase that can be exported on the events?